watertap/costing/watertap_costing_package.py

```diff
@@ -586,6 +629,67 @@ def cost_caoh2_mixer(blk):
             blk.lime_kg_per_day,
         )
 
+    @staticmethod
+    def cost_crystallizer(blk, cost_type=CrystallizerCostType.default):
+        """
+        Function for costing the FC crystallizer by the mass flow of produced crystals.
+
+        Args:
+            cost_type - Option for crystallizer cost function type - volume or mass basis
+        """
+        _make_capital_cost_var(blk)
+
+        if (
+            cost_type == CrystallizerCostType.default
+            or cost_type == CrystallizerCostType.mass_basis
+        ):
+            blk.capital_cost_constraint = pyo.Constraint(
+                expr=blk.capital_cost
+                == pyo.units.convert(
+                    (
+                        blk.costing_package.fc_crystallizer_iec_percent
+                        * blk.costing_package.fc_crystallizer_fob_unit_cost
+                        * (
+                            sum(
+                                blk.unit_model.solids.flow_mass_phase_comp[0, "Sol", j]
+                                for j in blk.unit_model.config.property_package.solute_set
+                            )
+                            / blk.costing_package.fc_crystallizer_ref_capacity
+                        )
+                        ** blk.costing_package.fc_crystallizer_ref_exponent
+                    ),
+                    to_units=blk.costing_package.base_currency,
+                )
+            )

```

Comment:
I've generally written different costing types as their own method (see `cost_pump` for example) for maximal flexibility. I.e. you can specify the method directly *or* use the `CrystallizerCostType` attribute.

Reply:
Done.